src/main/java/org/folio/circulation/domain/LoanRepository.java

```diff
@@ -51,11 +51,12 @@ public LoanRepository(Clients clients) {
       storageLoan.put("loanPolicyId", loanAndRelatedRecords.getLoanPolicy().getId());
     }
 
+    User user = loanAndRelatedRecords.getLoan().getUser();
     return loansStorageClient.post(storageLoan).thenApply(response -> {
       if (response.getStatusCode() == 201) {
         return succeeded(
           loanAndRelatedRecords.withLoan(Loan.from(response.getJson(),
-            loanAndRelatedRecords.getLoan().getItem())));
+            loanAndRelatedRecords.getLoan().getItem()).withUser(user)));
```

Comment:
Can you help me understand this change, why does the user need to be updated?

Reply:
When loan is saved, the **loan** field in **LoanAndRelatedRecords**  is updated and does not contain user representation. This representation is required to send patron notice just after loan is saved successfully.

Reply:
`Hmm, LoanAndRelatedRecords.withLoan(...` should not lose the user information that was there, I'd prefer this was fixed internally rather than here.

That might mean checking that the user for the loan is the same, and then keeping the existing user.

If that doesn't make sense, I can try to demonstrate it tomorrow.

Reply:
I'm not sure there is anything to fix here. I think the problem is that the loan is being recreated via the response of the `post` to loans-storage. That response will only have IDs. Maybe we should look up the user ID again or at least verify that the IDs match before we jam it into the loan.

If we are going to drop in the user, then we should probably just switch to this to build the loan object:
https://github.com/folio-org/mod-circulation/blob/a720b4667a6c9d9fa86accbc889025fefcbc37e0/src/main/java/org/folio/circulation/domain/Loan.java#L84-L88

Reply:
@mreno-EBSCO Thanks, yeah, I reckon using the extended from might be clearer.

It’s a minor point, I wondered if it might be clearer to have a dedicated method, to do the remembering of the item and user automatically.

Part of this is the compromises mod-circulation makes in how it models the domain, both the related records (mostly to carry state between async methods) and the retention of JSON representations (to avoid conversions).